src/test/java/teammates/test/driver/GmailServiceMaker.java

```diff
@@ -0,0 +1,120 @@
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.store.FileDataStoreFactory;
+import com.google.api.services.gmail.Gmail;
+import com.google.api.services.gmail.GmailScopes;
+
+import teammates.common.util.Const;
+
+/**
+ * Class that builds a Gmail service for use in Gmail API.
+ */
+final class GmailServiceMaker {
+
+    /** Global instance of the JSON factory. */
+    private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+
+    /** Global instance of the HTTP transport. */
+    private static final HttpTransport HTTP_TRANSPORT;
+
+    static {
+        try {
+            HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+        } catch (GeneralSecurityException | IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private final String username;
+    private final boolean shouldUseFreshCredentials;
+
+    GmailServiceMaker(String username) {
+        this(username, false);
+    }
+
+    GmailServiceMaker(String username, boolean shouldUseFreshCredentials) {
+        this.username = username;
+        this.shouldUseFreshCredentials = shouldUseFreshCredentials;
+    }
+
+    /**
+     * Build and return an authorized Gmail client service.
+     * @return an authorized Gmail client service
+     */
+    Gmail makeGmailService() throws IOException {
+        Credential credential = authorizeAndCreateCredentials();
+        return new Gmail.Builder(HTTP_TRANSPORT, JSON_FACTORY, credential)
+                .setApplicationName("teammates")
+                .build();
+    }
+
+    /**
+     * Authorizes the user and creates an authorized Credential.
+     * @return an authorized Credential
+     */
+    private Credential authorizeAndCreateCredentials() throws IOException {
+        GoogleClientSecrets clientSecrets = loadClientSecretFromJson();
+
+        GoogleAuthorizationCodeFlow flow = buildFlow(clientSecrets);
+
+        if (shouldUseFreshCredentials) {
+            flow.getCredentialDataStore().delete(username);
+        }
+
+        if (flow.getCredentialDataStore().get(username) == null) {
+            System.out.println("Please login as: " + username);
+        }
+
+        return getCredentialFromFlow(flow);
+    }
+
+    private GoogleClientSecrets loadClientSecretFromJson() throws IOException {
+        InputStream in;
+        try {
+            in = new FileInputStream(new File(TestProperties.TEST_GMAIL_API_FOLDER, "client_secret.json"));
+        } catch (FileNotFoundException e) {
+            throw new AssertionError("You need to set up your client_secret.json." + Const.EOL
+                    + "See Gmail API section in the documentation (development.md#deploying-to-a-staging-server)", e);
```

Comment:
But that is in the cause which is all the way below and the user will have to read almost the entire stack trace to see that. Won't it be better to state it early on?